File: Python/ex1.py
# Litkowski Norbert
# Cw.1
# Python

# McCulloch-Pitts neuron


def f(weights, inputs, m):
    x = float(0)
    for i in range(m):
        x += (weights[i] * inputs[i])
    if x >= 0:
        return 1
    else:
        return 0


def and_gate():
    inputs = []
    weights = [float(0.3), float(0.3), float(-0.5)]
    m = 3
    print('\nAND gate\n')
    x = input('Input 1: ')
    y = input('Input 2: ')
    if(x.isdigit() and y.isdigit()):
        inputs.append(float(int(x)))
        inputs.append(float(int(y)))
        inputs.append(float(1))
    else:
        print('\nBad input')
        return
    print("\nResult: {}".format(f(weights, inputs, m)))


def nand_gate():
    inputs = []
    weights = [float(-0.4), float(-0.4), float(0.6)]
    m = 3
    print('\nNAND gate\n')
    x = input('Input 1: ')
    y = input('Input 2: ')
    if(x.isdigit() and y.isdigit()):
        inputs.append(float(int(x)))
        inputs.append(float(int(y)))
        inputs.append(float(1))
    else:
        print('\nBad input')
        return
    print("\nResult: {}".format(f(weights, inputs, m)))


def or_gate():
    inputs = []
    weights = [float(0.3), float(0.3), float(-0.2)]
    m = 3
    print('\nOR gate\n')
    x = input('Input 1: ')
    y = input('Input 2: ')
    if(x.isdigit() and y.isdigit()):
        inputs.append(float(int(x)))
        inputs.append(float(int(y)))
        inputs.append(float(1))
    else:
        print('\nBad input')
        return
    print("\nResult: {}".format(f(weights, inputs, m)))


def not_gate():
    inputs = []
    weights = [float(-0.5), float(0.3)]
    m = 2
    print('\nNOT gate\n')
    x = input('Input: ')
    if(x.isdigit()):
        inputs.append(float(int(x)))
        inputs.append(float(1))
    else:
        print('\nBad input')
        return
    print("\nResult: {}".format(f(weights, inputs, m)))


def main():
    while True:
        print("\n1. NOT gate")
        print("2. AND gate")
        print("3. NAND gate")
        print("4. OR gate")
        print("0. Exit")
        x = input("\nChoice: ")
        try:
            x = int(x)
        except ValueError:
            print("\nBad input")
        if(x == 1):
            not_gate()
        elif(x == 2):
            and_gate()
        elif(x == 3):
            nand_gate()
        elif(x == 4):
            or_gate()
        elif(x == 0):
            break
        else:
            print("\nBad input...")


if __name__ == '__main__':
    main()

File: Julia/ex1.jl
# Ćwiczenia 1
# Model McCulloch-Pittsa

# Zad: Kożystając z modelu neurony M.-Pittsa, zaimplementować następujące
# bramki logiczne (znaleźć w_i(i <= i <= n)).
# 1) NOT
# 2) AND
# 3) NAND
# 4) OR

function dot(v1, v2, m)
    x = 0
    for i in 1:m
        x += v1[i] * v2[i]
    end
    return x
end

function f(weights, inputs, m)
    if dot(weights, inputs, m) >= 0
        return 1
    else return 0
    end
end

function not_gate()
    m = 2
    weights = [-0.5, 0.3]
    print("\033[92mNOT GATE \nInput: \033[0m")
    x = parse(Int64, readline())
    if isa(x, Number)
        inputs = [x, 1]
    else
        print("\nBad input")
        return
    end
    println(string("Result: ", f(weights, inputs, m)))
end

function and_gate()
    m = 3
    weights = [0.3, 0.3, -0.5]
    print("\033[92mAND GATE \nInput 1: \033[0m")
    x = parse(Int64, readline())
    print("\033[92mInput 2: ")
    y = parse(Int64, readline())
    if isa(x, Number)
        inputs = [x, y, 1]
    else
        print("\nBad input")
        return
    end
    println(string("Result: ", f(weights, inputs, m), "\033[0m"))
end

function nand_gate()
    m = 3
    weights = [-0.4, -0.4, 0.6]
    print("\033[95mNAND GATE \n\033[92mInput 1: \033[0m")
    x = parse(Int64, readline())
    print("\033[92mInput 2: ")
    y = parse(Int64, readline())
    if isa(x, Number)
        inputs = [x, y, 1]
    else
        print("\nBad input")
        return
    end
    println(string("Result: ", f(weights, inputs, m), "\033[0m"))
end

function or_gate()
    m = 3
    weights = [0.3, 0.3, -0.2]
    print("\033[95mOR GATE \n\033[92mInput 1: \033[0m")
    x = parse(Int64, readline())
    print("\033[92mInput 2: ")
    y = parse(Int64, readline())
    if isa(x, Number)
        inputs = [x, y, 1]
    else
        print("\nBad input")
        return
    end
    println(string("Result: ", f(weights, inputs, m), "\033[0m"))
end



# println(f(100))
# not_gate()
# and_gate()
# nand_gate()
or_gate()